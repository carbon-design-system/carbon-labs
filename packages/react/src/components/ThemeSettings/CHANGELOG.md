# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.6.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-theme-settings@0.5.0...@carbon-labs/react-theme-settings@0.6.0) (2025-05-28)


### Bug Fixes

* update exports to match UI shell ([#605](https://github.com/carbon-design-system/carbon-labs/issues/605)) ([1bf4682](https://github.com/carbon-design-system/carbon-labs/commit/1bf46822620b44cc1ad9ce58913bf26e9c3a2ca1))





# [0.5.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-theme-settings@0.4.0...@carbon-labs/react-theme-settings@0.5.0) (2025-05-08)


### Bug Fixes

* correct package exports ([#539](https://github.com/carbon-design-system/carbon-labs/issues/539)) ([ef7557e](https://github.com/carbon-design-system/carbon-labs/commit/ef7557e10c1385a6bdc60d32361ce7ba2dad263c))





# [0.4.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-theme-settings@0.3.0...@carbon-labs/react-theme-settings@0.4.0) (2025-04-04)

**Note:** Version bump only for package @carbon-labs/react-theme-settings





# [0.3.0](https://github.com/carbon-design-system/carbon-labs/compare/@carbon-labs/react-theme-settings@0.2.0...@carbon-labs/react-theme-settings@0.3.0) (2025-03-26)


### Features

* **uishell:** add header story with components update carbon/react ([#465](https://github.com/carbon-design-system/carbon-labs/issues/465)) ([a20ab0c](https://github.com/carbon-design-system/carbon-labs/commit/a20ab0c571bf739cbb906dfd08134c1966d06e0b))





# 0.2.0 (2025-03-19)


### Features

* ThemeSettings component with in context example ([#451](https://github.com/carbon-design-system/carbon-labs/issues/451)) ([9a0de11](https://github.com/carbon-design-system/carbon-labs/commit/9a0de11dd8c8be4ecfac329cbfc3695f89b761a6))





# 0.1.0 (2025-01-30)


### Features
