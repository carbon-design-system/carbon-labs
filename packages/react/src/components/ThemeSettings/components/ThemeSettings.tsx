/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import classNames from 'classnames';

// import './theme-settings.scss';
import { FormGroup } from '@carbon/react';
import { usePrefix } from '@carbon-labs/utilities/es/index.js';

const blockClass = `theme-settings`;

interface ThemeSettingsProps {
  children: React.ReactNode;
  className?: string;
  legendText?: string;
}

export const ThemeSettings = ({
  children,
  className,
  legendText = 'Theme settings',
  ...rest
}: ThemeSettingsProps) => {
  const prefix = usePrefix();

  return (
    <div
      className={classNames(className, `${prefix}--${blockClass}`)}
      {...rest}>
      <FormGroup
        legendText={legendText}
        className={`${prefix}--${blockClass}__form-group`}>
        {children}
      </FormGroup>
    </div>
  );
};

export default ThemeSettings;
