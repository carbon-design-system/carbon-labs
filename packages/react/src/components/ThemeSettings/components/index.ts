/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { ThemeSettings } from './ThemeSettings';
export { ThemeSwitcher } from './ThemeSwitcher';
export { ThemeSetDropdown } from './ThemeSetDropdown';
export { ThemeMenuComplement } from './ThemeMenuComplement';
export { themeSets } from './theme-settings-consts';
