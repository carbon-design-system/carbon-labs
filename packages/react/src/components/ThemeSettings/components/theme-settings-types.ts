/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
export type CarbonTheme = 'white' | 'g10' | 'g90' | 'g100';

export type ThemeSetType = 'white/g90' | 'g10/g90' | 'white/g100' | 'g10/g100';

export interface ThemeSetOption {
  text: string;
  value: ThemeSetType;
}

export type ThemeSettingType = 'light' | 'system' | 'dark';
