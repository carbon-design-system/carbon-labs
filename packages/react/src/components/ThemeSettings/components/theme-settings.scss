/**
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

@use '@carbon/react/scss/type';
@use '@carbon/react/scss/config';
@use '@carbon/react/scss/spacing' as *;

$prefix: 'clabs' !default;

.#{$prefix}--theme-settings__form-group {
  display: flex;
  flex-direction: column;
  gap: $spacing-05;
}
