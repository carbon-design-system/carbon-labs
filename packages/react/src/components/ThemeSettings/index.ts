/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
export {
  themeSets,
  ThemeSettings,
  ThemeSwitcher,
  ThemeSetDropdown,
  ThemeMenuComplement,
} from './components/index';

export { useThemeSetting } from './utils/use-theme-setting';
