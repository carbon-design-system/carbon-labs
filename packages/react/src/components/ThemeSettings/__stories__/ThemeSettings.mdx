import { ArgTypes, Canvas, Meta } from '@storybook/blocks';
import * as ThemeSettingsStories from './ThemeSettings.stories';

<Meta isTemplate />

# ThemeSettings

- **Initiative owner(s):** Lee Chase
- **Status:** Draft
- **Target library:** TBD
- **Target library maintainer(s) / PR Reviewer(s):** N/A
- **Support channel:** `#carbon-labs`

{/* <!-- START doctoc generated TOC please keep comment here to allow auto update --> */}
{/* <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE --> */}

> ðŸ’¡ Check our
> [Stackblitz](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/ThemeSettings)
> example implementation.

[![Edit carbon-labs](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/github/carbon-design-system/carbon-labs/tree/main/examples/react/ThemeSettings)

## Table of Contents

- [Overview](#overview)
- [Getting started](#getting-started)

{/* <!-- END doctoc generated TOC please keep comment here to allow auto update --> */}

## Overview

This is a Theme Settings component, the main part of which offers the options
`light`, `system` and `dark` to the user. If choosing `system` the user should
expect there current system settings to dictate the current theming.

<Canvas of={ThemeSettingsStories.InContextUseThemeSetting} />

## Getting started

### Install

```bash
yarn add @carbon/react
yarn add @carbon-labs/react-theme-switcher
```

### Components

- ThemeSettings: Container component for other ThemeSettings
- ThemeSwitcher: 3 way switch between light, system and dark ThemeSettings
- ThemeSetDropdown: Used if user is able to switch between theme sets
  `white/g90`, `white/g100`, `g10/g90` and `g10/g100`.
- ThemeMenuComplement: Used if user is permitted to toggle menu complement
  (often global menu is complement of main theme).

### Utilities

- useThemeSetting: Given a theme setting, theme set and a value for complement
  returns a Carbon theme based on these settings and users system preferences.

NOTE: Theme set can be a fixed value e.g. `white/g90` if ThemeSetDropdown is not
used.

### JS (via import)

```javascript
import { Theme } from '@carbon/react';
import {
  ThemeSettings,
  ThemeMenuComplement,
  ThemeSetDropdown,
  ThemeSwitcher,
  useThemeSetting,
} from '@carbon-labs/react-theme-switcher/es/index';

function App() {
  const [themeSetting, setThemeSetting] = useState(`system`);
  const [themeMenuComplement, setThemeMenuComplement] = useState(false);
  const [themeSet, setThemeSet] = useState('white/g100');
  const theme = useThemeSetting(themeSetting, themeSet, false);
  const themeHeader = useThemeSetting(
    themeSetting,
    themeSet,
    themeMenuComplement
  );

  return (
    <Theme className={'theme-setting-in-context'} theme={themeHeader}>
      <header className="theme-setting-in-context__header">
        A sample header
      </header>
      <Theme as="main" className="theme-setting-in-context__main" theme={theme}>
        <h2>Your app goes here.</h2>

        <ThemeSettings>
          <ThemeSwitcher
            onChange={setThemeSetting}
            value={themeSetting}></ThemeSwitcher>
          <ThemeMenuComplement
            id="theme-menu-complement"
            labelText="Complement menu theme"
            checked={themeMenuComplement}
            onChange={setThemeMenuComplement}
          />
          <ThemeSetDropdown
            id="theme-dropdown"
            label="Theme set"
            titleText="Theme set"
            value={themeSet}
            onChange={setThemeSet}
          />
        </ThemeSettings>
      </Theme>
    </Theme>
  );
}

export default App;
```

### SCSS

In your styles file import

```
@use '@carbon/react';
@use '@carbon-labs/react-theme-switcher/scss/theme-switcher';
```
