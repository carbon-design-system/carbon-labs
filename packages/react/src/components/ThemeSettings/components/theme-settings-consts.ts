/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */
import { ThemeSetOption } from './theme-settings-types';

export const themeSets: ThemeSetOption[] = [
  { text: 'White / G90', value: 'white/g90' },
  { text: 'G10 / G90', value: 'g10/g90' },
  { text: 'White / G100', value: 'white/g100' },
  { text: 'G10 / G100', value: 'g10/g100' },
];
