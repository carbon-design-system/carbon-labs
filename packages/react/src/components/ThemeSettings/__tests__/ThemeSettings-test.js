/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import '@testing-library/jest-dom';

import { ThemeSettings } from '../components/ThemeSettings';
import {
  ThemeMenuComplement,
  ThemeSetDropdown,
  ThemeSwitcher,
} from '../components';

jest.mock('./button.scss', () => ({}));
describe('ThemeSettings', () => {
  describe('renders as expected - Component API', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <ThemeSettings>
          <ThemeSwitcher></ThemeSwitcher>
          <ThemeSetDropdown
            id="theme-dropdown"
            label="Theme set"
            titleText="Theme set"
            value="g10/g100"
          />
          <ThemeMenuComplement
            id="theme-menu-complement"
            labelText="Complement menu theme"
            checked={false}
          />
        </ThemeSettings>
      );
      expect(container).toMatchSnapshot();
    });

    it('should render a content switcher', () => {
      render(
        <ThemeSettings>
          <ThemeSwitcher></ThemeSwitcher>
        </ThemeSettings>
      );

      expect(screen.getAllByRole('tab')).toHaveLength(3);
    });

    it('should render a one theme set dropdown', () => {
      render(
        <ThemeSettings>
          <ThemeSetDropdown
            id="theme-dropdown"
            label="Theme set"
            titleText="Theme set"
            value="g10/g100"
          />
        </ThemeSettings>
      );

      expect(screen.getByRole('combobox'));
    });

    it('should render a one theme complement checkbox', () => {
      render(
        <ThemeSettings>
          <ThemeMenuComplement
            id="theme-menu-complement"
            labelText="Complement menu theme"
            checked={false}
          />
        </ThemeSettings>
      );

      expect(screen.getByRole('checkbox'));
    });
  });

  describe('behaves as expected', () => {
    it('should change theme setting', async () => {
      const onChange = jest.fn();

      render(
        <ThemeSettings>
          <ThemeSwitcher onChange={onChange} value="system"></ThemeSwitcher>
        </ThemeSettings>
      );

      const tabs = screen.getAllByRole('tab');
      await userEvent.click(tabs[0]);

      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith('light');

      await userEvent.click(tabs[2]);
      expect(onChange).toHaveBeenLastCalledWith('dark');

      await userEvent.click(tabs[1]);
      expect(onChange).toHaveBeenLastCalledWith('system');
    });

    it('should change theme set dropdown', async () => {
      window.HTMLElement.prototype.scrollIntoView = function () {};

      const onChange = jest.fn();

      const { rerender } = render(
        <ThemeSettings>
          <ThemeSetDropdown
            id="theme-dropdown"
            label="Theme set"
            titleText="Theme set"
            value="g10/g100"
            onChange={onChange}
          />
        </ThemeSettings>
      );

      const combo = screen.getByRole('combobox');
      await userEvent.click(combo);
      await userEvent.click(screen.getByText('White / G90'));
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith('white/g90');

      await userEvent.click(combo);
      await userEvent.click(screen.getByText('White / G100'));
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith('white/g100');

      await userEvent.click(combo);
      await userEvent.click(screen.getByText('G10 / G90'));
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith('g10/g90');

      rerender(
        <ThemeSettings>
          <ThemeSetDropdown
            id="theme-dropdown"
            label="Theme set"
            titleText="Theme set"
            value="white/g100"
            onChange={onChange}
          />
        </ThemeSettings>
      );
      await userEvent.click(combo);
      await userEvent.click(screen.getByText('G10 / G100'));
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith('g10/g100');
    });

    it('should should change menu complement', async () => {
      const onChange = jest.fn();

      const { rerender } = render(
        <ThemeSettings>
          <ThemeMenuComplement
            id="theme-menu-complement"
            labelText="Complement menu theme"
            checked={false}
            onChange={onChange}
          />
        </ThemeSettings>
      );

      const check = screen.getByRole('checkbox');
      await userEvent.click(check);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenLastCalledWith(true);

      rerender(
        <ThemeSettings>
          <ThemeMenuComplement
            id="theme-menu-complement"
            labelText="Complement menu theme"
            checked={true}
            onChange={onChange}
          />
        </ThemeSettings>
      );

      await userEvent.click(check);
      expect(onChange).toHaveBeenLastCalledWith(false);
    });
  });
});
