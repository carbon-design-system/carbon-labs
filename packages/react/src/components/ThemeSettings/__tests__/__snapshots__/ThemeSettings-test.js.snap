// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSettings renders as expected - Component API should match snapshot 1`] = `
<div>
  <div
    class="clabs--theme-settings"
  >
    <fieldset
      class="cds--fieldset clabs--theme-settings__form-group"
    >
      <legend
        class="cds--label"
      >
        Theme settings
      </legend>
      <div
        class="cds--content-switcher theme-switcher cds--content-switcher--lg cds--layout--size-lg cds--content-switcher--icon-only cds--layout-constraint--size__default-md cds--layout-constraint--size__min-sm cds--layout-constraint--size__max-lg"
        role="tablist"
      >
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--content-switcher-popover__wrapper cds--icon-tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-label="Light"
              aria-selected="false"
              class="cds--content-switcher-btn cds--btn cds--btn--lg cds--layout--size-lg cds--btn--primary cds--btn--icon-only"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 1H8.5V3.5H7.5z"
                />
                <path
                  d="M10.8 3.4H13.3V4.4H10.8z"
                  transform="rotate(-45.001 12.074 3.922)"
                />
                <path
                  d="M12.5 7.5H15V8.5H12.5z"
                />
                <path
                  d="M11.6 10.8H12.6V13.3H11.6z"
                  transform="rotate(-45.001 12.077 12.075)"
                />
                <path
                  d="M7.5 12.5H8.5V15H7.5z"
                />
                <path
                  d="M2.7 11.6H5.2V12.6H2.7z"
                  transform="rotate(-45.001 3.929 12.078)"
                />
                <path
                  d="M1 7.5H3.5V8.5H1z"
                />
                <path
                  d="M3.4 2.7H4.4V5.2H3.4z"
                  transform="rotate(-45.001 3.926 3.925)"
                />
                <path
                  d="M8,6c1.1,0,2,0.9,2,2s-0.9,2-2,2S6,9.1,6,8S6.9,6,8,6 M8,5C6.3,5,5,6.3,5,8s1.3,3,3,3s3-1.3,3-3S9.7,5,8,5z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r1:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Light
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--content-switcher-popover__wrapper cds--icon-tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-label="System"
              aria-selected="false"
              class="cds--content-switcher-btn cds--btn cds--btn--lg cds--layout--size-lg cds--btn--primary cds--btn--icon-only"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 2H17V5H15zM27 15H30V17H27zM15 27H17V30H15zM2 15H5V17H2z"
                />
                <path
                  d="M6.22 5.73H8.219999999999999V8.73H6.22z"
                  transform="rotate(-45 7.227 7.236)"
                />
                <path
                  d="M23.27 6.23H26.27V8.23H23.27z"
                  transform="rotate(-45 24.766 7.232)"
                />
                <path
                  d="M23.77 23.27H25.77V26.27H23.77z"
                  transform="rotate(-45 24.77 24.77)"
                />
                <path
                  d="M5.47 25.13L7.59 23 9 24.42 6.88 26.54 5.47 25.13zM16 8a8 8 0 108 8A8 8 0 0016 8zm0 14a6 6 0 010-12z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r5:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              System
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
        <span
          class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--content-switcher-popover__wrapper cds--icon-tooltip"
        >
          <div
            class="cds--tooltip-trigger__wrapper"
          >
            <button
              aria-label="Dark"
              aria-selected="false"
              class="cds--content-switcher-btn cds--btn cds--btn--lg cds--layout--size-lg cds--btn--primary cds--btn--icon-only"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.2,2.3c-1,4.4,1.7,8.7,6.1,9.8c0.1,0,0.1,0,0.2,0c-1.1,1.2-2.7,1.8-4.3,1.8c-0.1,0-0.2,0-0.2,0C5.6,13.8,3,11,3.2,7.7 C3.2,5.3,4.8,3.1,7.2,2.3 M8,1L8,1C4.1,1.6,1.5,5.3,2.1,9.1c0.6,3.3,3.4,5.8,6.8,5.9c0.1,0,0.2,0,0.3,0c2.3,0,4.4-1.1,5.8-3 c0.2-0.2,0.1-0.6-0.1-0.7c-0.1-0.1-0.2-0.1-0.3-0.1c-3.9-0.3-6.7-3.8-6.4-7.6C8.3,3,8.4,2.4,8.6,1.8c0.1-0.3,0-0.6-0.3-0.7 C8.1,1,8.1,1,8,1z"
                />
              </svg>
            </button>
          </div>
          <span
            aria-hidden="true"
            class="cds--popover"
            id="tooltip-:r9:"
            role="tooltip"
          >
            <span
              class="cds--popover-content cds--tooltip-content"
            >
              Dark
            </span>
            <span
              class="cds--popover-caret"
            />
          </span>
        </span>
      </div>
      <div
        class="cds--dropdown__wrapper cds--list-box__wrapper theme-set-dropdown"
      >
        <label
          class="cds--label"
          for="downshift-:re:-toggle-button"
          id="downshift-:re:-label"
        >
          Theme set
        </label>
        <div
          class="cds--dropdown cds--list-box"
          id="theme-dropdown"
        >
          <button
            aria-activedescendant=""
            aria-controls="downshift-:re:-menu"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-:re:-label"
            class="cds--list-box__field"
            id="downshift-:re:-toggle-button"
            role="combobox"
            tabindex="0"
            title="G10 / G100"
            type="button"
          >
            <span
              class="cds--list-box__label"
            >
              G10 / G100
            </span>
            <div
              class="cds--list-box__menu-icon"
            >
              <svg
                aria-label="Open menu"
                fill="currentColor"
                focusable="false"
                height="16"
                name="chevron--down"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
                <title>
                  Open menu
                </title>
              </svg>
            </div>
          </button>
          <ul
            aria-labelledby="downshift-:re:-label"
            class="cds--list-box__menu"
            id="downshift-:re:-menu"
            role="listbox"
          />
        </div>
      </div>
      <div
        class="cds--form-item cds--checkbox-wrapper theme-menu-complement"
      >
        <input
          class="cds--checkbox"
          id="theme-menu-complement"
          type="checkbox"
        />
        <label
          class="cds--checkbox-label"
          for="theme-menu-complement"
          title=""
        >
          <span
            class="cds--checkbox-label-text"
            dir="auto"
          >
            Complement menu theme
          </span>
        </label>
        <div
          class="cds--checkbox__validation-msg"
        />
      </div>
    </fieldset>
  </div>
</div>
`;
